service/s3/s3manager/upload.go

```diff
@@ -617,6 +621,7 @@ func (u *multiuploader) upload(firstBuf io.ReadSeeker, firstPart []byte) (*Uploa
 		Location:  uploadLocation,
 		VersionID: complete.VersionId,
 		UploadID:  u.uploadID,
+		ETag: complete.ETag,
```

Comment:
The `ETag` should also be set when uploading a file using the upload manager's single part code path, `singlePart` method, line 486. The `ETag` value is available on the [PutObjectOutput.ETag](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#PutObjectOutput) member.